packages/generators/src/handlers/react.ts

```diff
@@ -0,0 +1,80 @@
+import path from "path";
+import { CustomGenerator } from "../types";
+import { questions as defaultQuestions } from "./default";
+const templatePath = path.resolve(__dirname, "../../init-template/react");
+const resolveFile = (file: string): string => {
+  return path.resolve(templatePath, file);
+};
+
+/**
+ * Asks questions including default ones to the user used to modify generation
+ * @param self Generator values
+ * @param Question Contains questions
+ */
+
+export async function questions(
+  self: CustomGenerator,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  Question: Record<string, any>,
+): Promise<void> {
+  await defaultQuestions(self, Question, true, {
+    devServer: true,
+    htmlWebpackPlugin: true,
+  });
+
+  // Add react dependencies
+  self.dependencies.push("react", "react-dom");
+
+  // Add webpack-dev-server always
+  self.dependencies.push("webpack-dev-server");
+
+  // Add html-webpack-plugin always
+  self.dependencies.push("html-webpack-plugin");
+
+  switch (self.answers.langType) {
+    case "Typescript":
+      self.dependencies.push("@types/react", "@types/react-dom");
+      break;
+    case "ES6":
+      self.dependencies.push("@babel/preset-react");
+      break;
+  }
+}
+
+/**
+ * Handles generation of project files
+ * @param self Generator values
+ */
+export function generate(self: CustomGenerator): void {
+  const files = ["./index.html", "./src/assets/webpack.png", "webpack.config.js", "package.json"];
+
+  switch (self.answers.langType) {
+    case "Typescript":
+      self.answers.entry = "./src/index.tsx";
+      files.push("tsconfig.json", "index.d.ts", "./src/App.tsx", self.answers.entry as string);
+      break;
+    case "ES6":
+      self.answers.entry = "./src/index.js";
+      files.push("./src/App.js", self.answers.entry as string);
+      break;
+  }
+
+  switch (self.answers.cssType) {
+    case "CSS only":
+      files.push("./src/styles/global.css");
+      break;
+    case "SASS":
+      files.push("./src/styles/global.scss");
+      break;
+    case "LESS":
+      files.push("./src/styles/global.less");
+      break;
+    case "Stylus":
+      files.push("./src/styles/global.styl");
+      break;
+  }
+
+  for (const file of files) {
+    self.fs.copyTpl(resolveFile(file + ".tpl"), self.destinationPath(file as string), self.answers);
+  }
+}
```

Comment:
We really should avoid duplication of code

Reply:
Well, these are two separate templates for very different use-cases and should be independent from each other. Coupling them for the sake of it shouldn't be a way to go, would lead to way more bugs in future.

Reply:
I am okay with separate templates, we can always make improvements based on user needs and feedback.

Reply:
/cc @alexander-akait

Reply:
I am okay, if other developers will support it :+1: we need more reviews

Reply:
/cc @anshumanv

Reply:
I'm okay with different handler because different templates will have different options, in fact I like how the default template is being reused here to minimize duplication.

Reply:
Closing this discussion now. Thanks everyone.